internal/app/backendapi/apisrv/apisrv.go

```diff
@@ -51,68 +49,33 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-// Logrus structured logging setup
-var (
-	beLogFields = log.Fields{
-		"app":       "openmatch",
-		"component": "backend",
-	}
-	beLog = log.WithFields(beLogFields)
-)
-
 // BackendAPI implements backend API Server, the server generated by compiling
 // the protobuf, by fulfilling the API Client interface.
 type BackendAPI struct {
-	grpc *grpc.Server
-	cfg  config.View
-	pool *redis.Pool
+	cfg    config.View
+	pool   *redis.Pool
+	logger *log.Entry
 }
-type backendAPI BackendAPI
 
 // New returns an instantiated srvice
-func New(cfg config.View, pool *redis.Pool) *BackendAPI {
-	s := BackendAPI{
-		pool: pool,
-		grpc: grpc.NewServer(grpc.StatsHandler(&ocgrpc.ServerHandler{})),
-		cfg:  cfg,
-	}
-
-	// Add a hook to the logger to auto-count log lines for metrics output thru OpenCensus
-	log.AddHook(metrics.NewHook(BeLogLines, KeySeverity))
-
-	pb.RegisterBackendServer(s.grpc, (*backendAPI)(&s))
-	beLog.Info("Successfully registered gRPC server")
-	return &s
-}
-
-// Open starts the api grpc service listening on the configured port.
-func (s *BackendAPI) Open() error {
-	ln, err := net.Listen("tcp", ":"+s.cfg.GetString("api.backend.port"))
-	if err != nil {
-		beLog.WithFields(log.Fields{
-			"error": err.Error(),
-			"port":  s.cfg.GetInt("api.backend.port"),
-		}).Error("net.Listen() error")
-		return err
+func New(omSrv *serving.OpenMatchServer) *BackendAPI {
```

Comment:
Will we be using the return value from New ever?  It seems the pattern is that it registers the rpc server, and the return value isn't used?  

If not, New doesn't seem the best name, but naming is hard and I don't have any good suggestions.  Register?  Start (though it isn't actually serving until start is called on the serving.openMatchServer, right?)?

Also, if the return value is never used, could it be backendAPI instead of BackendAPI?  Is there any reason to leak the type?


Here and for the other services too.